commands/run.go

```diff
@@ -0,0 +1,220 @@
+
+		if fileFlag == "" {
+			testsFile, err = loadTestFile(findTestFile())
+		} else {
+			testsFile, err = loadTestFile(fileFlag)
+		}
+
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			os.Exit(1)
+		}
+
+		results := map[string]error{}
+
+		for _, suite := range suites {
+			tests := testsFile.TestsToRun(suite)
+			loadedChecks, err := checks.LoadChecks(tests)
+			if err != nil {
+				fmt.Println("ERROR: Unable to load checks:", err)
+				continue
+			}
+
+			checksToRun := make(chan checks.CheckToRun, len(loadedChecks))
+			checkResults := make(chan checkResult, len(loadedChecks))
+
+			for i := 0; i < runtime.NumCPU(); i++ {
+				go runCheck(checksToRun, checkResults)
+			}
+
+			for _, c := range loadedChecks {
+				checksToRun <- c
+			}
+
+			for i := 0; i < len(loadedChecks); i++ {
+				result := <-checkResults
+				results[result.Name] = result.Result
+			}
+
+			close(checksToRun)
+			close(checkResults)
+		}
+
+		exitCode := 0
+
+		for name, result := range results {
+			if result == nil {
+				fmt.Printf("%s: SUCCESS\n", name)
+				continue
+			}
+
+			fmt.Printf("%s: FAILURE\n%s\n", name, result)
+			exitCode = 1
+		}
+
+		os.Exit(exitCode)
+	},
+}
+
+type checkResult struct {
+	Name   string
+	Result error
+}
+
+func runCheck(incomingChecks chan checks.CheckToRun, results chan checkResult) {
+	for check := range incomingChecks {
+		results <- checkResult{
+			Name:   check.Name,
+			Result: check.Check(),
+		}
+	}
+}
+
+func findTestFile() string {
+	possiblePaths := []string{
+		"tests.yml",
+		"tests.yaml",
+		"redalert.yml",
+		"redalert.yaml",
+		filepath.Join(os.Getenv("HOME"), ".redalert", "tests.yml"),
+		filepath.Join(os.Getenv("HOME"), ".redalert", "tests.yaml"),
+		filepath.Join(os.Getenv("APPDATA"), "redalert", "tests.yml"),
+		filepath.Join(os.Getenv("APPDATA"), "redalert", "tests.yaml"),
+		filepath.Join(os.Getenv("HOME"), ".redalert", "redalert.yml"),
+		filepath.Join(os.Getenv("HOME"), ".redalert", "redalert.yaml"),
+		filepath.Join(os.Getenv("APPDATA"), "redalert", "redalert.yml"),
+		filepath.Join(os.Getenv("APPDATA"), "redalert", "redalert.yaml"),
+		"/etc/redalert/tests.yml",
+		"/etc/redalert/tests.yaml",
+		"/etc/redalert/redalert.yml",
+		"/etc/redalert/redalert.yaml",
+		"C:\\redalert\\tests.yml",
+		"C:\\redalert\\tests.yaml",
+		"C:\\redalert\\redalert.yml",
+		"C:\\redalert\\redalert.yaml",
+	}
+
+	for _, path := range possiblePaths {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			return path
+		}
+	}
+
+	return ""
+}
+
+func loadTestFile(path string) (testfile.TestFile, error) {
+	if path == "" {
+		return testfile.TestFile{}, errors.New("no tests files found")
+	}
+
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return testfile.TestFile{}, err
+	}
+
+	var tf testfile.TestFile
+
+	err = yaml.Unmarshal(content, &tf)
+	if err != nil {
+		return testfile.TestFile{}, err
+	}
+
+	dir := filepath.Dir(path)
+	content, err = ioutil.ReadFile(filepath.Join(dir, "aliases.yml"))
```

Comment:
Could this use the same pattern as the tests.(yml|yaml) pattern with a possiblePaths[] array? I find that more readable but your call.